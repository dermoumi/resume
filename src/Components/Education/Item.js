import React, { Component, Fragment } from 'react'
import styled from 'styled-components'

const StyleBase = styled.article`
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .title {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.5rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }

  .period {
    font-weight: 400;
    margin-left: 8px;
    white-space: nowrap;
    color: ${props => props.theme.accent};
  }

  .period-arrow {
    font-family: icomoon;
    font-size: 0.9em;
  }
`

export default class Item extends Component {
  render() {
    let period
    if (
      Array.isArray(this.props.period) &&
      typeof this.props.period[0] === 'string'
    ) {
      period = (
        <Fragment>
          {this.props.period[0]}
          <span className="period-arrow"> &#xe900; </span>
          {this.props.period[1]}
        </Fragment>
      )
    } else {
      period = this.props.period
    }

    return (
      <StyleBase {...this.props}>
        <div className="heading">
          <div className="title">{this.props.degree}</div>
          <div className="sub-title">
            {this.props.info} <span className="period">{period}</span>
          </div>
        </div>
      </StyleBase>
    )
  }
}
