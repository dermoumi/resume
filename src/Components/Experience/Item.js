import React, { Component, Fragment } from 'react'
import styled from 'styled-components'

const StyleBase = styled.article`
  margin-bottom: 16px;
  /* max-width: 608px; */

  &:last-child {
    margin-bottom: 0;
  }

  .label {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.25rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }

  .label a,
  .sub-title a {
    color: inherit;
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }

    &:after {
      font-family: icomoon;
      font-size: 0.9em;
      content: '\\e907';
      margin-left: 5px;

      @media print {
        display: none;
      }
    }
  }

  .period {
    font-weight: 400;
    white-space: nowrap;
    color: ${props => props.theme.accent};
  }

  .period-arrow {
    font-family: icomoon;
    font-size: 0.9em;
  }

  .item-content {
    line-height: 1.25rem;
    padding-top: 4px;

    ul {
      list-style: none;
      margin: 0;
      padding: 0;
      padding-left: calc(1em + 10px);

      li {
        margin-bottom: 6px;

        &:before {
          font-family: icomoon;
          content: '\\e906';
          color: ${props => props.theme.accent};
          margin-left: calc(-1em - 10px);
          margin-right: 10px;
        }
      }
    }
  }
`

export default class Item extends Component {
  render() {
    let period
    if (Array.isArray(this.props.period)) {
      period = (
        <Fragment>
          {this.props.period[0]}
          <span className="period-arrow"> &#xe900; </span>
          {this.props.period[1]}
        </Fragment>
      )
    } else {
      period = this.props.period
    }

    const tasks =
      this.props.tasks &&
      this.props.tasks.map((task, i) => (
        <li key={i} dangerouslySetInnerHTML={{ __html: task }} />
      ))

    return (
      <StyleBase {...this.props}>
        <div className="heading">
          <div className="label">
            {this.props.labelLink ? (
              <a
                href={this.props.labelLink}
                target="_blank"
                rel="noreferrer noopener"
              >
                {this.props.label}
              </a>
            ) : (
              this.props.label
            )}
          </div>
          <div className="sub-title">
            {this.props.subtitleLink ? (
              <a
                href={this.props.subtitleLink}
                target="_blank"
                rel="noreferrer noopener"
              >
                {this.props.subtitle}
              </a>
            ) : (
              this.props.subtitle
            )}{' '}
            <span className="period">{period}</span>
          </div>
        </div>
        {tasks && (
          <div className="item-content">
            <ul>{tasks}</ul>
          </div>
        )}
      </StyleBase>
    )
  }
}
