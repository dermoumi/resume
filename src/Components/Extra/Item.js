import React, { Component } from 'react'
import styled from 'styled-components'

const StyleBase = styled.article`
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .label {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.25rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }
`

export default class Item extends Component {
  render() {
    return (
      <StyleBase {...this.props}>
        <div className="heading">
          <div className="label">{this.props.label}</div>
          <div className="sub-title">{this.props.details}</div>
        </div>
      </StyleBase>
    )
  }
}
