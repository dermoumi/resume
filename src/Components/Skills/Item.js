import React, { Component } from 'react';
import styled from 'styled-components';

const StyleBase = styled.article`
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .title {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.5rem;
    color: ${props => props.theme.primary};
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ${props => props.theme.secondary};
  }

  .level {
    font-weight: 400;
    margin-left: 4px;
    font-size: 0.8em;
    letter-spacing: 0;
    white-space: nowrap;
    color: ${props => props.theme.accent};
  }
`;

export default class Item extends Component {
  render() {
    return (<StyleBase {...this.props}>
      <div className="heading">
        <div className="title">{this.props.label} <span className="level">{this.props.level}</span></div>
        <div className="sub-title">{this.props.details}</div>
      </div>
    </StyleBase>);
  }
}
