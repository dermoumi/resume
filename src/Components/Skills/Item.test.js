import React from 'react'
import { mount, shallow } from 'enzyme'

import Item from './Item'

describe('Skills section item', () => {
  it('renders title', () => {
    const title = 'My title'
    const wrapper = shallow(<Item label={title} />)

    expect(wrapper.text()).toContain(title)
  })

  it('renders details', () => {
    const details = 'My details'
    const wrapper = shallow(<Item details={details} />)

    expect(wrapper.text()).toContain(details)
  })

  it('renders level', () => {
    const level = 'My level'
    const wrapper = shallow(<Item level={level} />)

    expect(wrapper.text()).toContain(level)
  })

  it('respects the theme', () => {
    const theme = {
      primary: '#FFCC00',
      secondary: '#00FFCC',
      accent: '#CC00FF',
    }

    const props = {
      label: 'My title',
      details: 'My details',
      level: 'My level',
      theme,
    }

    const wrapper = mount(<Item {...props} />)
    expect(wrapper).toMatchSnapshot()

    expect(wrapper).toHaveStyleRule('color', theme.primary, {
      modifier: '.label',
    })

    expect(wrapper).toHaveStyleRule('color', theme.secondary, {
      modifier: '.sub-title',
    })

    expect(wrapper).toHaveStyleRule('color', theme.accent, {
      modifier: '.level',
    })
  })
})
