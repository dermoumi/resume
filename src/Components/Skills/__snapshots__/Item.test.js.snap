// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills section item respects the theme 1`] = `
.c0 {
  margin-bottom: 16px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c0 .label {
  font-size: 1.1em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.25rem;
  color: #FFCC00;
}

.c0 .sub-title {
  font-size: 0.9em;
  line-height: 1.25rem;
  text-transform: uppercase;
  color: #00FFCC;
}

.c0 .level {
  font-weight: 400;
  font-size: 0.8em;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  white-space: nowrap;
  line-height: 1.5rem;
  color: #CC00FF;
}

<Item
  details="My details"
  label="My title"
  level="My level"
  theme={
    Object {
      "accent": "#CC00FF",
      "primary": "#FFCC00",
      "secondary": "#00FFCC",
    }
  }
>
  <styled.article
    details="My details"
    label="My title"
    level="My level"
    theme={
      Object {
        "accent": "#CC00FF",
        "primary": "#FFCC00",
        "secondary": "#00FFCC",
      }
    }
  >
    <StyledComponent
      details="My details"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  margin-bottom: 16px;

  &:last-child {
    margin-bottom: 0;
  }

  .label {
    font-size: 1.1em;
    letter-spacing: 1px;
    line-height: 1.25rem;
    color: ",
              [Function],
              ";
  }

  .sub-title {
    font-size: 0.9em;
    line-height: 1.25rem;
    text-transform: uppercase;
    color: ",
              [Function],
              ";
  }

  .level {
    font-weight: 400;
    font-size: 0.8em;
    letter-spacing: 0;
    white-space: nowrap;
    line-height: 1.5rem;
    color: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.article",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "article",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      label="My title"
      level="My level"
      theme={
        Object {
          "accent": "#CC00FF",
          "primary": "#FFCC00",
          "secondary": "#00FFCC",
        }
      }
    >
      <article
        className="c0"
        label="My title"
      >
        <div
          className="heading"
        >
          <div
            className="label"
          >
            My title
             
            <span
              className="level"
            >
              My level
            </span>
          </div>
          <div
            className="sub-title"
          >
            My details
          </div>
        </div>
      </article>
    </StyledComponent>
  </styled.article>
</Item>
`;
